lib/p2p/Peer.ts

```diff
@@ -120,24 +106,45 @@ class Peer extends EventEmitter {
   }
 
   public getStatus = (): string => {
+    let status: string;
     if (this.connected) {
-      return `Connected to peer (${this.id})`;
+      status = this.nodePubKey ?
+        `Connected to peer ${this.nodePubKey}` :
+        `Connected pre-handshake to peer ${addressUtils.toString(this.socketAddress)}`;
     } else {
-      return 'Not connected';
+      status = 'Not connected';
     }
+    return status;
   }
 
-  public open = async (handshakeData: HandshakeState): Promise<void> => {
+  /**
+   * Prepare a connection for use by ensuring it is active, exchanging [[HelloPacket]] with handshake data,
+   * and emit the `open` event if everything succeeds. Throw an error on unexpected handshake data.
+   * @param handshakeData our handshake data to send to the peer
+   * @param nodePubKey the expected nodePubKey of the node we are opening a connection with
+   */
+  public open = async (handshakeData: HandshakeState, nodePubKey?: string): Promise<void> => {
     assert(!this.opened);
     this.opened = true;
 
+    if (this.closed) {
+      throw 'wtf';
```

Comment:
Good error message :rofl:

Reply:
Haha, I put this in as a sanity test for the sanity tests when I was trying to see why they'd never terminate. I'll probably change this to something a little more formal - probably just an assert like above - but it's really something that should never happen.